unittests/internaltests.py

```diff
@@ -1715,3 +1717,475 @@ def test_option_key_from_string(self) -> None:
+                    DEB_HOST_ARCH_CPU=ppc64el
+                    DEB_HOST_ARCH_ENDIAN=little
+                    DEB_HOST_ARCH_LIBC=gnu
+                    DEB_HOST_ARCH_OS=linux
+                    DEB_HOST_GNU_CPU=powerpc64le
+                    DEB_HOST_GNU_SYSTEM=linux-gnu
+                    DEB_HOST_GNU_TYPE=powerpc64le-linux-gnu
+                    DEB_HOST_MULTIARCH=powerpc64le-linux-gnu
+                    '''
+                ),
+                '',
+                {'PATH': '/usr/bin'},
+                MachineInfo.populated(
+                    compilers={
+                        'c': ['/usr/bin/powerpc64le-linux-gnu-gcc'],
+                        'cpp': ['/usr/bin/powerpc64le-linux-gnu-g++'],
+                        'objc': ['/usr/bin/powerpc64le-linux-gnu-gobjc'],
+                        'objcpp': ['/usr/bin/powerpc64le-linux-gnu-gobjc++'],
+                    },
+                    binaries={
+                        'ar': ['/usr/bin/powerpc64le-linux-gnu-ar'],
+                        'strip': ['/usr/bin/powerpc64le-linux-gnu-strip'],
+                        'objcopy': ['/usr/bin/powerpc64le-linux-gnu-objcopy'],
+                        'ld': ['/usr/bin/powerpc64le-linux-gnu-ld'],
+                        'cmake': ['/usr/bin/cmake'],
+                        'pkg-config': ['/usr/bin/powerpc64le-linux-gnu-pkg-config'],
+                        'cups-config': ['/usr/bin/cups-config'],
+                    },
+                    properties={
+                    },
+                    compile_args={
+                    },
+                    link_args={
+                    },
+                    cmake={
+                        'CMAKE_C_COMPILER': ['/usr/bin/powerpc64le-linux-gnu-gcc'],
+                        'CMAKE_CXX_COMPILER': ['/usr/bin/powerpc64le-linux-gnu-g++'],
+                        'CMAKE_SYSTEM_NAME': 'Linux',
+                        'CMAKE_SYSTEM_PROCESSOR': 'ppc64le',
+                    },
+                    system='linux',
+                    subsystem='linux',
+                    kernel='linux',
+                    # TODO: Currently ppc64, but native builds have ppc64le
+                    # https://github.com/mesonbuild/meson/issues/13741
+                    cpu='TODO',
+                    cpu_family='ppc64',
+                    endian='little',
+                ),
+            ),
+        ]:
+            with self.subTest(title), \
+                    unittest.mock.patch.dict('os.environ', env, clear=True), \
+                    unittest.mock.patch('mesonbuild.scripts.env2mfile.locate_path') as mock_locate_path:
+                mock_locate_path.side_effect = locate_path
+                options = argparse.Namespace()
+                options.gccsuffix = gccsuffix
+                actual = to_machine_info(dpkg_arch, options)
+
+                if expected.system == 'TODO':
+                    print('TODO: %s: system() -> %s' % (title, actual.system))
```

Comment:
We don't use the `%` formatters, either f-strings for str.format. In this case I think: `f'TODO: {title}: system() -> {actual.system}'` increases readability too

Reply:
Clearly I spend too much time in ancient container environments (or ancient codebases) where f-strings are either nonexistent, or considered new and suspicious :-)